### PR TITLE
[WIP] 1.12.0

### PR DESCRIPTION
**Pull request by GeoffreyBooth**
@jashkenas @lydell I think we’re ready for 1.12.0.

Besides the updated `lib` folder, browser compiler, docs, annotated source and change log, I snuck in a few more documentation improvements:

- [x] Docs examples with quotation marks are fixed. (Try running `tim.age + " " + tim.height` on the current http://coffeescript.org.) Now `Cakefile` [properly escapes the quotes](http://stackoverflow.com/a/1082558/223225).
- [x] Eligible HTML entities (basically, anything other than `&`, `<` or `>`) are now written as unicode characters, [a recommended practice](http://stackoverflow.com/a/436637/223225).
- [x] Along those lines, quotation marks in regular text are now curly. Ellipses when used as text and not as splat code are now ellipses characters.
- [x] A link to the browser tests (`test.html`) was added under Resources.

### CODE REVIEW

File: docs/v1/annotated-source/coffee-script.html [hunk 1]
```diff
@@ -193,6 +190,27 @@
               <div class="pilwrap ">
                 <a class="pilcrow" href="#section-5">&#182;</a>
               </div>
+              <p>The contents of a <script> block are encoded via UTF-16, so if any extended
```
Comment (vendethiel) [inline docs/v1/annotated-source/coffee-script.html]: Shouldn't `<script>` use escaped <> here?

Comment (vendethiel) [inline docs/v1/annotated-source/coffee-script.html]: I guess docco lets html fall through...

**Comment by greghuc** [comment]
@GeoffreyBooth I believe 1.12.0 will be affected by the just-reported ['unescaped backticks in interpolated strings'](https://github.com/jashkenas/coffeescript/issues/4380) bug, though only if someone uses tagged template literals. I'll have a fix in the next few days.

File: docs/v1/index.html [hunk 2]
```diff
@@ -2594,6 +2806,88 @@
...
+          and with Markdown.
+        </li>
+        <li>
+          Within single-backtick embedded JavaScript, backticks can now be
+          escaped via <code>\`</code>.
+        </li>
+        <li>
+          CoffeeScript’s <a href="#objects_and_arrays">shorthand object
+          notation</a> is finally documented.
+        </li>
```
Comment (lydell) [inline docs/v1/index.html]: In my opinion, all the documentation or build system improvement points (such as this one) are just noise in the change log. It’s really awesome work, but not needed to be in the change log. Again, that’s _my_ opinion. :)

**Comment by lydell** [comment]
I agree that the 'unescaped backticks in interpolated strings' bug has to be fixed before a release.

File: docs/v1/annotated-source/coffee-script.html (hunk 1 above)
Comment (GeoffreyBooth) [inline docs/v1/annotated-source/coffee-script.html]: Wrapping the HTML in backticks fixes this.

**Comment by GeoffreyBooth** [comment]
Updated with #4383. How about now?

**Comment by lydell** [comment]
Everything looks good to me! 👍 Nice work.

**Comment by GeoffreyBooth** [comment]
Thanks. I don't have access to NPM. @lydell, can you deploy to there and then accept this PR?

**Comment by lydell** [comment]
Done.

**Comment by greghuc** [comment]
@GeoffreyBooth and everyone else involved - congrats on getting 1.12 out. Looking forward to Coffeescript 2 :-)

**Comment by GeoffreyBooth** [comment]
Thanks!

**Pull request merged**